tests/mock_requests.py

```diff
@@ -43,6 +45,19 @@ def raise_for_status(self):
             raise requests.exceptions.RequestException(self.error_msg)
 
 
+def intercept_text(text):
+    try:
+        if '10.1103/PhysRevD.89.084044' in text:
+            # replace with wrong DOI
+            text = text.replace('PhysRevD', 'INVALIDDOI')
+    except TypeError:
```

Comment:
Actually I can't reproduce the case in which the `try` is needed. In any case using a `str('PhysRevD')` or something similar would also be an option.

Reply:
I don't think this has to do with unicode, but with bytes versus strings. `Response.text` in request is a property, so it can't be reassigned. To make the `ONLINE` version work, I modify the `r._content` variable, which is bytes. If you have an alternative, I am interested.

Reply:
OK, I see. My point was that there was maybe a simple way to avoid duplicating the body of the function in the catch block. But there is probably none and it is fine like that for testing code.